spec/jobs/part_replication_audit_job_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+describe PartReplicationAuditJob, type: :job do
+  let(:cm) { create(:complete_moab, version: 2) }
+  let(:job) { described_class.new(cm, endpoint) }
+  let(:endpoint) { cm.zipped_moab_versions.first.zip_endpoint }
+  let(:logger) { instance_double(Logger) }
+
+  before do
+    allow(Audit::CatalogToArchive).to receive(:logger).and_return(logger)
+  end
+
+  describe '#queue_as' do
+    it 'builds the queue name' do
+      expect(job.queue_name).to eq("part_audit_#{endpoint.endpoint_name}")
+    end
+
+    context 'with a different endpoint' do
+      let(:endpoint) { create(:zip_endpoint, endpoint_name: 'north_foobar') }
+
+      it 'adapts the queue name' do
+        expect(job.queue_name).to eq('part_audit_north_foobar')
+      end
+    end
+  end
+
+  describe '#perform' do
+    let(:zmv1) { cm.zipped_moab_versions.first }
+    let(:zmv2) { cm.zipped_moab_versions.second }
+    let(:results) { job.send(:new_results, cm) }
+
+    it 'only checks parts for one endpoint' do
+      other_ep = create(:zip_endpoint)
+      other_zmv = cm.zipped_moab_versions.find_by(zip_endpoint: other_ep)
+      count = cm.zipped_moab_versions.where(zip_endpoint: endpoint).count
+      expect(job).to receive(:check_child_zip_part_attributes).with(ZippedMoabVersion, AuditResults).exactly(count).times
+      expect(job).not_to receive(:check_child_zip_part_attributes).with(other_zmv, AuditResults)
```

Comment:
this one, i'm a little murkier on...

so `other_ep` should auto-link to the default pres policy because of how its factory is defined... but since no backfilling is happening here (right?), i don't think i'd expect there to be any `ZMV`s linked to `other_ep` anyway at this point, and so i wouldn't expect `job` to get `check_child_zip_part_attributes` called on `other_zmv` anyway.

actually, i somewhat suspect that `other_zmv` is `nil` in this case, and that if it weren't (like if the backfill had happened), it'd be a 2 element array, since `cm` is on version 2 (and so i'd also think that a `where` would be more appropriate than a `find_by` on line 34).

Reply:
discussed F2F, this works fine because:
1.`cm` isn't instantiated till it's referred to on line 34, and then the backfill happens automatically through AR hooks (because the new zip endpoint does alredy exist, due to line 33).
2. `find_by` will just return the first result if there's more than one result, and we only need one counter-example to show that the things we expect not to be checked are in fact filtered.